hovercontrol/src/hoverkite.rs

```diff
@@ -182,17 +181,29 @@ fn print_response(response: &Option<SideResponse>) {
 
 fn read_port(
     port: &mut Box<dyn SerialPort>,
-    buffer: &mut VecDeque<u8>,
-    side: Side,
+    buffer: &mut SliceDeque<u8>,
 ) -> Result<Option<SideResponse>, eyre::Report> {
     if port.bytes_to_read()? > 0 {
         let mut temp = [0; 100];
         let bytes_read = port.read(&mut temp)?;
         buffer.extend(&temp[0..bytes_read]);
     }
+    // I guess we could make parse() return Result<(response, len), ...>
+    // to avoid this byte-at-a-time nonsense.
+    for len in 1..=buffer.len() {
+        match SideResponse::parse(&buffer[..len]) {
```

Comment:
```
SideResponse::parse(&[u8]) -> nb::Result<(SideResponse, usize), ParseError>
SideResponse::parse_exact(&[u8]) -> nb::Result<SideResponse, ParseError>

enum ProtocolError {
    MessageTooLong,
    InvalidSide(u8),
    InvalidCommand(u8),
    UnexpectedByte(u8),
    // ??? ReadError, // from the impl Read
}
```

Reply:
I did the ProtocolError bit of this, but the refactor required to implement parse_exact() is not something that I want to do in this PR (or possibly ever: see below). Having everything working in terms of Command and Response, and the serialization code looking similar is more important than whether this particular piece of laptop-side code is performant. In the last couple of PRs, I feel like I've spent a lot of time wandering around making cosmetic changes to parsing code without really understanding what clean no_std serialization/deserialization code looks like, because I've not examined any other implementations to see what's possible.

What I really want to do is import postcard/binfmt/whatever and `#[derive(Serialize, Deserlialize]` on DirectedMessage and SideResponse, and then delete our hand-rolled .write_to() and .parse() implementations entirely. I don't think that parse_exact() is a step in that direction, so I'm going to leave it as it is. If my postcard spike goes nowhere then we can revisit .parse_exact() and tidy this up then.